--- src/nethttpd-for-netcgi1/nethttpd_kernel.ml.orig	2007-11-01 22:12:31.000000000 +0000
+++ src/nethttpd-for-netcgi1/nethttpd_kernel.ml	2009-02-04 21:39:19.000000000 +0000
@@ -249,9 +249,9 @@
 		  transfer_encoding <- `Identity;
 	  );
 	  (* Update the header: *)
-	  ( match transfer_encoding with
-	      | `Identity -> resp_header # delete_field "Transfer-Encoding"
-	      | `Chunked  -> set_transfer_encoding resp_header ["chunked", []]
+	  ( match transfer_encoding, suppress_body with
+	      | `Identity,false | _,true -> resp_header # delete_field "Transfer-Encoding"
+	      | `Chunked,false -> set_transfer_encoding resp_header ["chunked", []]
 	  );
 	  resp_header # delete_field "Trailer";
 	  set_date resp_header (Unix.time());
